### PR TITLE
Add documentation for cred store common values

### PR DESCRIPTION
**Pull request by simo5**
Add docs to explain what cred store values can be used and how

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -28,6 +28,7 @@ straight into the :doc:`high-level API documentation <gssapi>`.
 
    gssapi.rst
    gssapi.raw.rst
+   credstore.rst
```
[the remark this answers is not in the document]
Comment (simo5) [reply docs/source/index.rst]: ok

File: docs/source/credstore.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+client_keytab
+"""""""""""""
+
+The `client_keytab` keyword can be used in a credential store when it is used
+with the :func:`gssapi.raw.ext_cred_store.acquire_cred_from` /
+:func:`gssapi.raw.ext_cred_store.add_cred_from` functions, to indicate a
+custom location for a keytab containing client keys.
+It is not used in the context of calls used to store credentials.
+The value is a string in the form "type:residual" where type can be any
```
[the remark this answers is not in the document]
Comment (simo5) [reply docs/source/credstore.rst]: residual is the technical name as found in MIT docs, at most I can use "path" if you think residual is to be avoided.

**Comment by simo5** [comment]
@DirectXMan12 PTAL

**Comment by simo5** [comment]
@DirectXMan12 ping ?

File: docs/source/credstore.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (DirectXMan12) [reply docs/source/credstore.rst]: maybe put in in parens or something.  It's non-obvious to most people that haven't already read the MIT docs, but it's good to have it there for people who have.

**Comment by DirectXMan12** [comment]
responded to the `type:residual` comment, otherwise looks good

**Pull request merged**